docs/development/CodingStyle.md

```diff
@@ -0,0 +1,253 @@
+
+Check .map file to make sure no conversions sneak in, and use -Wdouble-promotion warning for the compiler
+
+Instead of sin() and cos(), use sin_approx() and cos_approx() from common/math.h.
+
+Float constants should be defined with "f" suffix, like 1.0f and 3.1415926f, otherwise double conversion might occur.
+
+#Functions
+
+##Naming
+Methods that return a boolean should be named as a question, and should not change any state. e.g. 'isOkToArm()'.
+
+Methods should have verb or verb-phrase names, like deletePage or save. Tell the system to 'do' something 'with' something. e.g. deleteAllPages(pageList).
+
+Non-static functions should be prefixed by their class. Eg baroUpdate and not updateCompass .
+
+Groups of functions acting on an 'object' should share the same prefix, e.g.
+```
+float biQuadFilterApply(...);
+void biQuadFilterInit(...);
+boolean biQuadIsReady();
+```
+rather than
+```
+float applyBiQuadFilter(...);
+void newBiQuadLpf(...);
+boolean isBiQuadReady();
+```
+
+##Parameter order
+Data should move from right to left, as in memcpy(void *dst, const void *src, size_t size).
+This also mimics the assignment operator (e.g. dst = src;)
+
+When a group of functions act on an 'object' then that object should be the first parameter for all the functions, e.g.:
+```
+float biQuadFilterApply(biquad_t *state, float sample);
+void biQuadNewLpf(biquad_t *state, float filterCutFreq, uint32_t refreshRate);
+```
+rather than
+```
+float biQuadFilterApply(float sample, biquad_t *state);
+void biQuadNewLpf(float filterCutFreq, biquad_t *state, uint32_t refreshRate);
+```
+
+##Declarations
+Functions not used outside their containing .c file should be declared static (or STATIC_UNIT_TESTED so they can be used in unit tests).
+
+Non-static functions should have their declaration in a single .h file.
+
+Don't make more than necessary visible for other modules, not even types. Pre-processor macros may be used to declare module internal things that must be shared with the modules test code but otherwise hidden.
+
+In the .h file:
+```
+#ifdef MODULENAME_INTERNALS_
+… declarations …
+#endif
+```
+In the module .c file, and in the test file but nowhere else, put "#define MODULENAME_INTERNALS_" just before including the .h file.
+
+##Implementation
+Keep functions short and distinctive.
+Think about unit test when you define your functions. Ideally you should implement the test cases before implementing the function.
+
+Never put multiple statements on a single line.
+Never put multiple assignments on a single line.
+Never put multiple assignments in a single statement.
+
+Defining constants using pre-processor macros is not preferred.
+Const-correctness should be enforced.
+This allows some errors to be picked up at compile time (for example getting the order of the parameters wrong in a call to memcpy).
+
+The pre-increment is preferred over the post-increment operator, i.e.  ++i rather than i++.
```

Comment:
there are 1012 lines with post-increment / post-decrement and 47 lines  with pre-increment/pre-decrement ..

This is IMO difference between C and C++, overloaded pre-increment is easier to optimize. Quite common idiom in C is `char* p = buf; *p++ = 1; *p++ = 2;`, where pre-increment form won't work.